### PR TITLE
Geo location

### PR DESCRIPTION
**Pull request by thohee**
the index commit has one unintential modification of a version number
db:migrate works
db:create could not be tested

the kiez restriction commit has been tested:
e.g. businesses in potsdam are not returned when asking for businesses in berlin
however, the 10km may be too narrow.

### CODE REVIEW

File: app/controllers/api/businesses_controller.rb [hunk 1]
```diff
@@ -19,8 +19,20 @@ def show
       @owner = @business.owner
     end
 
+    DistanceInKm = 10.0 # could be an api parameter
+    KMperDegree = 111.045
+    DistanceLatInDegrees = DistanceInKm / KMperDegree
```
Comment (djmaze) [inline app/controllers/api/businesses_controller.rb]: That's better. But in idiomatic Ruby, constants go to the top of a class and are written in uppercase letters only. I.e. `DISTANCE_IN_KM`

Comment (thohee) [reply app/controllers/api/businesses_controller.rb]: please improve implementation as required when merging

**Comment by thohee** [comment]
i suggest we add the index not too late because adding an index to a table which already contains a lot of entries may challenge the database.

We can wait with the kiez restriction until we have about 1000 businesses.
For know it is nice to zoom out to Germany and see all.
We could also add an option in the frontend to optionally restrict to x kilometers close to current location.

The suggested implementation with a squared kiez is the fastest you can get, but only with the index.
A circular kiez is not required, I think.

**Comment by djmaze** [comment]
@thohee Can you look at the latest changes in order to see if they make sense, from a geospatial standpoint?

**Comment by djmaze** [comment]
> Hence we have to use the average latitude (min_lat + max_lat / 2).

Thanks, will change that!

> Do we really need the 2km additional range?

Not necessarily. But it helps smoothen the experience a bit if users just move the map a bit.

> The fallback case does not test for not_rejected!

Good catch, will change.

> If we limit all to 100 we may get 100 from Hamburg when we query in Berlin. So the fallback is not very useful.

Yes, but we probably don't want to use the fallback _at all_ if we have more than 100 businesses. That would make API responses too big.

**Comment by djmaze** [comment]
Took out the limit on the legacy API. Still need feedback concerning the right calculation @thohee

**Comment by thohee** [comment]
Ok, if the two where conditions result in a combined sql statement.
The fix for scaling the additional longitude has not been commited yet.
If we merge the db migration to staging like this, it will probably not be executed, because many more recent migrations exist.

**Comment by djmaze** [comment]
From my perspective, this could be merged.

**Comment by thohee** [comment]
finally correct

**Pull request merged**